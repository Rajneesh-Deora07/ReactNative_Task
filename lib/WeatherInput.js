import React, {useState} from 'react';
import Strings from './localization';
import {
  StyleSheet,
  View,
  TextInput,
  Text,
  TouchableHighlight,
  Dimensions,
} from 'react-native';

const WeatherInput = (props) => {
  const [zipCode, setZipCode] = useState('');

  const fetchWeatherData = () => {
    props.fetchData(zipCode);
  };

  return (
    <View style={styles.containerStyle}>
      <Text>{Strings.how}</Text>
      <Text>{'Enter Zip Code'}</Text>
      <TextInput
        style={styles.inputStyle}
        value={zipCode}
        onChangeText={(text) => setZipCode(text)}
        keyboardType={'number-pad'}
        clearButtonMode={'while-editing'}
      />
      <TouchableHighlight onPress={fetchWeatherData}>
        <View style={styles.buttonStyle}>
          <Text>Check Weather</Text>
        </View>
      </TouchableHighlight>
    </View>
  );
};

const styles = StyleSheet.create({
  buttonStyle: {
    alignItems: 'center',
    backgroundColor: '#DDDDDD',
    padding: 10,
    marginTop: 10,
  },
  buttonTitleStyle: {
    fontSize: 20,
    color: 'white',
  },
  inputStyle: {
    height: 40,
    paddingLeft: 20,
    borderWidth: 2,
    borderColor: 'grey',
  },
  containerStyle: {
    marginTop: 50,
    width: Dimensions.get('window').width - 60,
    paddingLeft: 30,
    paddingRight: 30,
    height: 120,
  },
});

export default WeatherInput;
