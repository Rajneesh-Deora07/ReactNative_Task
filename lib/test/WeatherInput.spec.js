import React from 'react';
import {shallow} from 'enzyme';
import WeatherInput from '../WeatherInput';
import {View, TextInput, Text, TouchableHighlight} from 'react-native';
describe('WeatherInput test cases ------', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <WeatherInput />,
      //wrapper.setState({zipCode: '122002'}),
    );
  });
  it('Snapshot match', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('Render UI of weatherInput', () => {
    expect(wrapper.find(View).length).toBe(2);
    expect(wrapper.find(Text).length).toBe(3);
    expect(wrapper.find(TouchableHighlight).length).toBe(1);
    expect(wrapper.find(TextInput).length).toBe(1);
  });
});
