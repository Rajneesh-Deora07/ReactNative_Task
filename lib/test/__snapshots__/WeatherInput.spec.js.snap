// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherInput test cases ------ Snapshot match 1`] = `ShallowWrapper {}`;
